src/arrays/referencing_array_and_its_elements/hard/q3/MainTest.java

```diff
@@ -0,0 +1,103 @@
+package arrays.referencing_array_and_its_elements.hard.q3;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.utils.ArrayUtil.*;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public static int n = 10;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = arrayToInput(generateRandomArray(-100, 100, n));
+        return new Clause[]{
+                new StringLiteral("Enter " + n + " positive integers:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("The array with only even numbers is:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder(),
+                new Optional(new StringLiteral(" ")),
+        };
+    }
+
+    public void runMain() {
+        Even.main(new String[0]);
+    }
+
+    public static int[] getEven(int[] arr) {
```

Comment:
Change to something like,
int[] evens = new int[n];
int evenCount = 0;
if even
     events[evenCount++] = passedArray[x]
return copy of array using arrayCopy or something

to get a different solution code than the question